openlibrary/templates/book_providers/wikisource_download_options.html

```diff
@@ -0,0 +1,33 @@
+$def with(wikisource_id)
+
+$code:
+    # pardon this stupid way to avoid importing re in a html file
+    def is_langcode(string):
+        return all('a' <= char <= 'z' for char in string) and len(string) > 0
+
+    def split_id(string):
+        chunks = string.split(":")
+        if len(chunks) >= 2 and is_langcode(chunks[0]):
+            return chunks[0], ":".join(chunks[1:])
+        return 'en', string
+
+    langcode, title = split_id(wikisource_id)
+
+    def direct_url(type):
+        return 'https://ws-export.wmcloud.org/?format=' + type + '&lang=' + langcode + '&page=' + title
```

Comment:
```suggestion
    def direct_url(type):
        params = {"format": type, "lang": langcode, "page": title}
        base_url = "https://ws-export.wmcloud.org/"
        return "%s?%s" % (base_url, urlencode(params))
```
I spoke with Drini, and to be safe, let's error on the side of caution and URL encode this.